app/controllers/staff/activity_forms_controller.rb

```diff
@@ -282,24 +281,22 @@ def country_delivery_partners
     params.require(:activity).permit(country_delivery_partners: []).fetch("country_delivery_partners", []).reject(&:blank?)
   end
 
-  def planned_start_date
-    planned_start_date = params.require(:activity).permit(:planned_start_date)
-    {day: planned_start_date["planned_start_date(3i)"], month: planned_start_date["planned_start_date(2i)"], year: planned_start_date["planned_start_date(1i)"]}
-  end
-
-  def planned_end_date
-    planned_end_date = params.require(:activity).permit(:planned_end_date)
-    {day: planned_end_date["planned_end_date(3i)"], month: planned_end_date["planned_end_date(2i)"], year: planned_end_date["planned_end_date(1i)"]}
-  end
-
-  def actual_start_date
-    actual_start_date = params.require(:activity).permit(:actual_start_date)
-    {day: actual_start_date["actual_start_date(3i)"], month: actual_start_date["actual_start_date(2i)"], year: actual_start_date["actual_start_date(1i)"]}
+  def extract_date_parts_for(date_attr)
+    date_params = params.require(:activity).permit(date_attr)
+    {day: date_params["#{date_attr}(3i)"], month: date_params["#{date_attr}(2i)"], year: date_params["#{date_attr}(1i)"]}
   end
 
-  def actual_end_date
-    actual_end_date = params.require(:activity).permit(:actual_end_date)
-    {day: actual_end_date["actual_end_date(3i)"], month: actual_end_date["actual_end_date(2i)"], year: actual_end_date["actual_end_date(1i)"]}
+  def validate_inputs_on_dates_step
```

Comment:
Can we call this `assign_inputs_on_dates_step` instead to better reflect what this method is doing?

Reply:
Done, thank you for the review and for suggesting a better name!